modules/sdk-core/test/unit/rangeproof.ts

```diff
@@ -0,0 +1,106 @@
+import * as sinon from 'sinon';
+import * as paillierBigint from 'paillier-bigint';
+import Ecdsa from '../../src/account-lib/mpc/tss/ecdsa/ecdsa';
+import * as rangeProof from '../../src/account-lib/mpc/tss/ecdsa/rangeproof';
+import { NTilde } from '../../src/account-lib/mpc/tss/ecdsa/types';
+import { OpenSSL } from '../../src/openssl';
+
+describe('MtA range proof', function () {
+  let switchPrime = false;
+  let safePrimeMock: sinon.SinonStub;
+
+  let paillierKeyPair: paillierBigint.KeyPair;
+  let ntilde: NTilde;
+
+  before('set up paillier and ntile', async function () {
+    safePrimeMock = sinon.stub(OpenSSL.prototype, 'generateSafePrime').callsFake(async (bitlength: number) => {
```

Comment:
Stubbing `rangeProof.generateSafePrimes()` is more straightforward, but it doesn't work with node 14 for some reason.

Reply:
`generateSafePrime` doesn't work on node 14 unit tests. Are you sure ur stubs are being called then?